weather_mv/loader_pipeline/util.py

```diff
@@ -70,18 +71,53 @@ def to_json_serializable_type(value: t.Any) -> t.Any:
     return value
 
 
-def _only_target_vars(ds: xr.Dataset, data_vars: t.Optional[t.List[str]] = None) -> xr.Dataset:
+def _only_target_coordinate_vars(ds: xr.Dataset, is_merged_dataset: bool, data_vars: t.List[str]) -> t.List[str]:
+    """If the user specifies target fields in the dataset, get all the matching coords & data vars."""
+
+    # If the dataset is not the merged dataset (created for normalizing grib's schema),
+    # return the target fields specified by the user.
+    if not is_merged_dataset:
+        return data_vars
+
+    keep_coords_vars = []
+
+    for dv in data_vars:
+        keep_coords_vars.extend([v for v in ds.data_vars if v.endswith('_'+dv) or v.startswith(dv+'_')])
+        keep_coords_vars.extend([v for v in COORD_DISTRACTOR_SET if v in dv])
+
+    return keep_coords_vars
+
+
+def _only_target_vars(ds: xr.Dataset, is_merged_dataset: bool, data_vars: t.Optional[t.List[str]] = None) -> xr.Dataset:
     """If the user specifies target fields in the dataset, create a schema only from those fields."""
 
     # If there are no restrictions on data vars, include the whole dataset.
     if not data_vars:
         logger.info(f'target data_vars empty; using whole dataset; size: {ds.nbytes}')
         return ds
 
-    assert all([dv in ds.data_vars or dv in ds.coords for dv in data_vars]), 'Target variable must be in original ' \
-                                                                             'dataset. '
+    if not is_merged_dataset:
+        assert all([dv in ds.data_vars or dv in ds.coords for dv in data_vars]), 'Target variable must be in original '\
+                                                                                'dataset. '
+
+        dropped_ds = ds.drop_vars([v for v in ds.data_vars if v not in data_vars])
+    else:
+        drop_vars = []
+        check_target_variable = []
+        for dv in data_vars:
+            searched_data_vars = [x.endswith('_'+dv) or x.startswith(dv+'_') for x in ds.data_vars]
+            searched_coords = [] if dv not in COORD_DISTRACTOR_SET else [dv in ds.coords]
+            check_target_variable.append(searched_data_vars.__contains__(True) or searched_coords.__contains__(True))
```

Comment:
Maybe: 
```suggestion
            check_target_variable.append(any(searched_data_vars) or any(searched_coords))
```

However, can you explain what you're trying to do here? I'm a bit confused.

Reply:
I think you may want to use a `filter` and test if the list is empty? Let's talk about it.